pkg/oauth/server/osinserver/defaults.go

```diff
@@ -24,6 +24,7 @@ func NewDefaultServerConfig() *osin.ServerConfig {
 	config.AllowClientSecretInParams = true
 	config.AllowGetAccessRequest = true
 	config.RedirectUriSeparator = ","
+	config.ErrorStatusCode = http.StatusInternalServerError
```

Comment:
Shouldn't this be: StatusBadRequest as per: https://tools.ietf.org/html/rfc6749#section-5.2?